gpii/node_modules/settingsHandlers/src/settingsHandlerUtilities.js

```diff
@@ -193,16 +193,47 @@ gpii.settingsHandlers.checkRereadSettings = function (that) {
     }
 };
 
+/**
+ * Called to invoke settings handlers. Based on the first verifySettings option found in the payload,
+ * the settingshandler will be invoked with retrying enabled (if true) or without it (false).
+ * Structure of 'payload' argument is:
+ * {
+ *    "some.app.id": [{
+ *         "settings": { ... },
+ *         "options": {
+ *             ...,
+ *            "verifySettings": true
+ *         }
+ *    }]
+ *  }
+ * @param resolvedName {String} The resolved "trunk name" of the settings handler to be invoked - e.g. gpii.windows.registrySettingsHandler
+ * @param payload {Object} The full payload that would be sent to the SET method of the settings handler
+ * @param retryOptions {Object} (optional) Options governing the retry behaviour of the wrapped handler. This contains fields:
+ *     numRetries {Integer} A positive integer holding the total number of retries to make
+ *     retryInterval {Integer} A positive integer holder the delay in milliseconds between each retry
+ * @return a promise that will yield the original payload of the invoked SET method. In the case
+ *     that retrying is enabled, a rejection occurs if the GET payloads never match after the end of the nominated retry period
+ */
+gpii.settingsHandlers.dispatchSettingsHandler = function (resolvedName, payload, retryOptions) {
+    for (var appData in payload) {
+        appData = payload[appData];
+        if (appData[0].options && appData[0].options.verifySettings === true) {
```

Comment:
if(fluid.get(appData, [0, "options", "verifySettings"]) === true)

Also - add a TODO here that references the utility "gpii.lifecycleManager.specToSettingsHandler" as the one responsible for this strange layout - all of this infrastructure will have to be updated in step